### PR TITLE
Improve logging of negative btsieve health check

### PR DESCRIPTION
**Pull request by luckysori**
As @tcharding pointed out, the logs generated by cnd's health check on btsieve knew too much about how btsieve is implemented. This tries to fix that by leveraging the Problem Details JSON object returned with unsuccessful responses to btsieve.

Cool finding:

Functions that expect a closure as an argument also accept blocks that return a closure:

```rust
.map({
    let foo = self.foo.clone();
    move |_| println!("Foo: {}", foo)
})
```

This is particularly useful when working with futures.

### CODE REVIEW

File: cnd/src/btsieve/client.rs [hunk 1]
```diff
@@ -234,26 +235,45 @@ impl BtsieveHttpClient {
...
+                    move |mut response| {
+                        if response.status().is_success() {
+                            log::info!("Btsieve running at {}", endpoint);
+                        } else {
+                            let _ = response
+                                .json()
+                                .map({
+                                    let endpoint = endpoint.clone();
+                                    move |problem: HttpApiProblem| match problem {
+                                        HttpApiProblem {
```
Comment (luckysori) [inline cnd/src/btsieve/client.rs]: hadouken-indentation

File: cnd/src/btsieve/client.rs [hunk 2]
```diff
@@ -234,26 +235,48 @@ impl BtsieveHttpClient {
...
-                    if response.status().is_success() {
-                        log::info!("Btsieve running at {}", endpoint)
-                    } else {
-                        log::warn!(
-                            "Version of btsieve at {} does not match expected version",
-                            endpoint
-                        )
-                    }
+                .map_err(HealthCheckError::Reqwest)
+                .and_then(|response| HealthCheckResponse {
```
Comment (D4nte) [inline cnd/src/btsieve/client.rs]: I don't understand why you wrapped it in a Health Check Response

Comment (D4nte) [reply cnd/src/btsieve/client.rs]: Ah ok, it's because you wrap it in a Future.

**Comment by tcharding** [comment]
> Cool finding:
> 
> Functions that expect a closure as an argument also accept blocks that return a closure:
> 
> ```rust
> .map({
>     let foo = self.foo.clone();
>     move |_| println!("Foo: {}", foo)
> })
> ```

Isn't that semantically the same as writing this?

```rust
.map(move |_| {
    let foo = self.foo.clone();
    println!("Foo: {}", foo)
})
```

**Comment by thomaseizinger** [comment]
> > Cool finding:
> > Functions that expect a closure as an argument also accept blocks that return a closure:
> > ```rust
> > .map({
> >     let foo = self.foo.clone();
> >     move |_| println!("Foo: {}", foo)
> > })
> > ```
> 
> @luckysori isn't that semantically the same as writing this?
> 
> ```rust
> .map(move |_| {
>     let foo = self.foo.clone();
>     println!("Foo: {}", foo)
> })
> ```

Nope! Not quite :)
The block is executed immediately and the closure is returned!

**Comment by thomaseizinger** [comment]
I tweeted about it here because I am so excited about this :D
https://twitter.com/oetzn/status/1166912187174023169

**Pull request merged**